VKappProject/VKappProject/Controllers/FriendListTableViewController.swift

```diff
@@ -14,60 +15,76 @@ final class FriendListTableViewController: UITableViewController {
 
     // MARK: - Private Properties
 
-    private let users: [User] = [
-        User(
-            avatarImageName: Constants.UserImageNames.bryleeCollinsImageName,
-            name: Constants.UserNames.bryleeCollinsName
-        ),
-        User(avatarImageName: Constants.UserImageNames.emilieRivasImageName, name: Constants.UserNames.emilieRivasName),
-        User(avatarImageName: Constants.UserImageNames.connorLloydImageName, name: Constants.UserNames.connorLloydName),
-        User(avatarImageName: Constants.UserImageNames.ireneNormanImageName, name: Constants.UserNames.ireneNormanName),
-        User(
-            avatarImageName: Constants.UserImageNames.kamronThomasImageName,
-            name: Constants.UserNames.kamronThomasName
-        ),
-        User(
-            avatarImageName: Constants.UserImageNames.kaylynnGrossImageName,
-            name: Constants.UserNames.kaylynnGrossName
-        ),
-        User(avatarImageName: Constants.UserImageNames.kyleRosalesImageName, name: Constants.UserNames.kyleRosalesName),
-        User(avatarImageName: Constants.UserImageNames.martinSteinImageName, name: Constants.UserNames.martinSteinName)
-    ]
-
-    private var sections: [Character: [User]] = [:]
+    private var users: [UserItem] = []
+    private var sections: [Character: [UserItem]] = [:]
     private var sectionTitles: [Character] = []
+    private let service = VKAPIService()
 
     // MARK: - LifeCycle
 
     override func awakeFromNib() {
         super.awakeFromNib()
-        groupingUser()
+        setupUI()
     }
 
     // MARK: - Public Method
 
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         guard segue.identifier == Constants.Identifiers.friendCollectionViewControllerIdentifier,
-              let friendCollectionViewController = segue.destination as? FriendCollectionViewController else { return }
-        guard let indexPath = tableView.indexPathForSelectedRow else { return }
-        let keys = sections.keys.sorted()
-        let key = keys[indexPath.section]
-        guard let friendAvatar = sections[key]?[indexPath.row] else { return }
-        friendCollectionViewController.friendAvatarName = friendAvatar.avatarImageName
+              let friendCollectionViewController = segue.destination as? FriendCollectionViewController,
+              let indexPath = tableView.indexPathForSelectedRow,
+              let id = getOneUser(indexPath: indexPath)?.id else { return }
+        friendCollectionViewController.id = String(id)
     }
 
     // MARK: - Private Method
 
+    private func setupUI() {
+        fetchFriendRequest()
+    }
+
+    private func setupFriends() {
+        DispatchQueue.main.async { [self] in
```

Comment:
```suggestion
        DispatchQueue.main.async { [weak self] in
```
Ммм... weak тут разве не нужен?

Reply:
Я считаю что нет